### ISSUE TITLE
for custom callbacks in module fit, training script can not finish

### ISSUE THREAD

**Issue by Davidrjx** [apache/incubator-mxnet#9897]
## Description

use (batch&eval end)callback kwargs for tensorflow-tensorboard writer, but after two epoch, training process did not terminate, namely in pending state ,and require CTRL+C to stop 
## Environment info (Required)
```
----------Python Info----------
Version      : 3.5.3
Compiler     : GCC 6.3.0 20170118
Build        : ('default', 'Jan 19 2017 14:11:04')
Arch         : ('64bit', '')
------------Pip Info-----------
Version      : 9.0.1
Directory    : ddd
----------MXNet Info-----------
Version      : 1.1.0
Directory    : ddddd
Commit Hash   : 07a83a0325a3d782513a04f47d711710972cb144
----------System Info----------
Platform     : Linux-4.9.0-4-amd64-x86_64-with-debian-9.3
system       : Linux
node         : ddddd
release      : 4.9.0-4-amd64
version      : #1 SMP Debian 4.9.65-3+deb9u1 (2017-12-23)
----------Hardware Info----------
machine      : x86_64
processor    :
Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                56
On-line CPU(s) list:   0-55
Thread(s) per core:    2
Core(s) per socket:    14
Socket(s):             2
NUMA node(s):          2
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 79
Model name:            Intel(R) Xeon(R) CPU E5-2680 v4 @ 2.40GHz
Stepping:              1
CPU MHz:               1207.763
CPU max MHz:           3300.0000
CPU min MHz:           1200.0000
BogoMIPS:              4799.93
Virtualization:        VT-x
L1d cache:             32K
L1i cache:             32K
L2 cache:              256K
L3 cache:              35840K
NUMA node0 CPU(s):     0-13,28-41
NUMA node1 CPU(s):     14-27,42-55
Flags:                 fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf eagerfpu pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch epb intel_pt tpr_shadow vnmi flexpriority ept vpid fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm cqm rdseed adx smap xsaveopt cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local dtherm ida arat pln pts
----------Network Test----------
Setting timeout: 10
Timing for Gluon Tutorial(cn): https://zh.gluon.ai, DNS: 1.4324 sec, LOAD: 3.8097 sec.
Timing for PYPI: https://pypi.python.org/pypi/pip, DNS: 0.1308 sec, LOAD: 0.8327 sec.
Timing for MXNet: https://github.com/apache/incubator-mxnet, DNS: 0.3458 sec, LOAD: 4.1076 sec.
Timing for Gluon Tutorial(en): http://gluon.mxnet.io, DNS: 1.3146 sec, LOAD: 1.2678 sec.
Timing for FashionMNIST: https://apache-mxnet.s3-accelerate.dualstack.amazonaws.com/gluon/dataset/fashion-mnist/train-labels-idx1-ubyte.gz, DNS: 0.4631 sec, LOAD: 3.4307 sec.
Timing for Conda: https://repo.continuum.io/pkgs/free/, DNS: 0.1860 sec, LOAD: 0.7889 sec.


```

Package used (Python/R/Scala/Julia):
I'm using Python

## Error Message:
(Paste the complete error message, including stack trace.)
![image](https://user-images.githubusercontent.com/4284791/36717966-0e1cd8bc-1bdb-11e8-8a59-f15a2450ac07.png)

## Minimum reproducible example

mxnet minist digits recongnition as my examples

## Steps to reproduce

1. python virtual env, install mxnet, tensorflow & tensorboard
2. code and run minist digits recongnition model training with two epochs

## What have you tried to solve it?

1. comment custom callbacks in module fit methods , it is valid but indeed i require those cbs for writing tensorboard events

**Comment by roywei** [comment]
@sandeep-krishnamurthy please kindly add label: Python, Data Visualization, thanks!

**Comment by sandeep-krishnamurthy** [comment]
@reminisce - Any suggestions?

**Comment by reminisce** [comment]
It looks like one logging thread got stuck.
The dmlc/tensorboard was adapted from TensorFlow. There is some gap between the adapted version and the original code in TensorFlow. We are working on a complete visualization solution bundled in MXNet. Stay tuned.

**Comment by Davidrjx** [comment]
@reminisce @sandeep-krishnamurthy @roywei thanks!
first of all, what i need to clarify was that i used MXNET with callable callback  to log events where tf's FileWirter and Summary were used in custom SummaryWriter class , and the custom class was invoked in `__call__` in callable callback
@reminisce your saying that one logging thread got stuck should be right, as above exception  was not there when i closed custom SummaryWriter in __call__ after calling callback every time. Or above exception occured once i placed close method of tf's Filewriter into close/`__del__` method of custom class at the time close of  tf's Filewriter got stuck via debugger.
look forward to the solution you mentioned

**Comment by piyushghai** [comment]
@reminisce Have we been able to address this issue in the visualization tool in DMLC ?
